src/historywork/WriteVerifiedCheckpointHashesWork.h

```diff
@@ -78,6 +86,13 @@ class WriteVerifiedCheckpointHashesWork : public BatchWork
     void startOutputFile();
     void endOutputFile();
     std::shared_ptr<std::ofstream> mOutputFile;
-    std::string mOutputFileName;
+    std::optional<std::filesystem::path> const mTrustedHashPath;
+    std::filesystem::path mOutputPath;
+    // If true, mOutputPath == mTrustedHashPath, and output
+    // will be written to a temporary file before being renamed to
+    // mOutputPath when verificaiton is complete.
+    bool mAppendToFile = false;
```

Comment:
I think we should always write to a temp file. That way, in the case of a crash or a failed verification, we don't output a broken trusted hash file. This also allows you to remove the flag and additional logic around `mAppendToFile`